cluster/node.go

```diff
@@ -75,18 +117,150 @@ func (n *Node) SetState(s NodeState) {
 	n.Unlock()
 }
 
-// provide thread safe json Marshaling
-func (n *Node) MarshalJSON() ([]byte, error) {
+func (n *Node) SetPartitions(part []int32) {
```

Comment:
oh wow I hadn't thought about on-the-fly adjustments like that. sounds good.
just so we're clear though:
* carbon input partition is set by the operator in the config, and is just something that represents how they manually shard their data, based on their relaying setup and how they route carbon traffic. until we have config reloading or an api for this, it can only be changed by restarting MT
* the mdm input does not seem to have a provision either to dynamically become aware of more partitions and call SetPartitions accordingly. is this something that still has to be implemented? (perhaps as a future improvement)